addons/crm/tests/common.py

```diff
@@ -162,9 +162,9 @@ def setUpClass(cls):
             'zip': '97648',
         })
 
-    def _create_leads_batch(self, lead_type='lead', count=10, email_dup_count=0,
-                            partner_count=0, partner_ids=None, user_ids=None,
-                            country_ids=None):
+    def _create_leads_data_batch(self, lead_type='lead', count=10, email_dup_count=0,
```

Comment:
Instead of changing the method, maybe simply add a "phone_numbers" argument that is None by default and when specified, is used to fill in crm.lead data by cycling through the provided numbers?

Very much like "partner_ids".

(You would have to test the existence of the phone_numbers variable to skip the assignment based on passed countries too but I think it's doable.)